packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1303,13 +1303,17 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   /// Implementations of this method should end with a call to the inherited
   /// method, as in `super.dispose()`.
   ///
-  /// ## Application shutdown
+  /// ## Caveats
   ///
-  /// This method is _not_ invoked when the application shuts down, because
-  /// there is no way to predict when that will happen. For example, a user's
-  /// battery could catch fire, or the user could drop the device into a
-  /// swimming pool, or the operating system could unilaterally terminate the
-  /// application process due to memory pressure.
+  /// This method is _not_ invoked at times where a developer might otherwise
+  /// expect it.
```

Comment:
This seems like an overly broad statement. Can we specify what these cases are where one might expect this to be called, but it isn't?

Reply:
Maybe this just needs to verbally link to the next two sections to indicate that they talk about two common cases of this?

Reply:
Added another clause here, wdyt?